### PR TITLE
feat: added code for account trust score badges

### LINKED ISSUE #9: AAU, I see the user account badges depending on their global trust scores
As a User,
I see any user's account badges depending on their global trust scores
So that end-users can see my reputation according to the MetaMask pre-set trust graph

----

## Technical notes:

* Use the "read" endpoints for "TrustScore" at the API level:
   * For our frontend to get the current trust score for a given DID
   * `GET /global-trust-scores/{DID}`
* When displaying the Account Profile page, display a badge depending on the trust score for this profile
* The mapping `Score <> Badge` is described in the acceptance criteria
* Badges and accuracy is given in the payload posted by Karma3

## Badges

- Users can have 2 types of badges: `Developer`, `Auditor` (cf. `trustScoreScope` field in the file payload)
- Only "Software Security" `trustScoreScope` will imply tiers
- For "Software Security" `trustScoreScope`, the tiers can be declined in 4 levels as described by the `accuracy` field in the file payload:
   - `Basic` (🥉Tier 3)
   - `Proficient` (🥈Tier 2)
   - `Distinguish` (🥇Tier 1)
   - No tier (Nothing is displayed)
- For "Software Development" `trustScoreScope` there is no tiers, but there is an unique badge
- A `result` field describes the report state of the peer (cf. value below)

Existing Badges:

### Badges mapping

> Percentage is described in the `accuracy` field of the file payload
> If `accuracy` is empty => No tier/badge
> If `trustScoreScope` is not "Snaps Security" => No tier/badge
> If `result` is -1 => No tier/badge

> Accuracy value is displayed as a tooltip when hovering the badge

- accuracy >= 0.99 (🥇 Tier 1 (**Distinguish**))
- 0.99 > accuracy >= 0.90 (🥈 Tier 2 (**Proficient**))
- 0.90 > accuracy >= 0.75 (🥉 Tier 3 (**Basic**))
- 0.75 > accuracy (no tier)

### Reported peers
👹 Reported actor: `result` field is -1

## Out of scope:

* Compute the score and upload the result to our API (done by Karma3 Lab)

## Acceptance Criteria:

- [ ] Given that I navigate to user profiles, I see their badges
- [ ] Given that hovering over a auditor badge displays a tooltip with the auditor accuracy in percentage

## Definition of Ready
- [X] Figma design ready
- [X] Acceptance criteria ready
- [X] Technical grooming done

## Mockup design

=> [FIGMA](https://www.figma.com/file/uHZRCcB6UCI1eX7dBjMEKK/SPD-Proof-of-Concept?type=design&node-id=90%3A4851&mode=design&t=1t2n31mTU7FmxkmW-1)

Positive badges

Negative badge: Reported user

### PR DESCRIPTION
**Pull request by satyajeetkolhapure**
## Description

Updated account badges component to receive trust score from api and show corresponding badges

image

### Related ticket

Fixes #9 

### Type of change

- [ ] Chore
- [ ] Bug fix
- [x] New feature
- [ ] Documentation update

### CODE REVIEW

**Review by alainncls — APPROVED** [review]

**Review by stanleyyconsensys — CHANGES_REQUESTED** [review]
LGTM, some comment need to verify on your side , and i would suggest move the business logic / API call in account role tag / account info to a hook if we have time

File: src/features/account/components/AccountRoleTags.tsx [hunk 1]
```diff
@@ -1,64 +1,123 @@
...
+    } else {
+      const role: AccountRole = {
+        roleType: AccountRoleType.Reported,
+        tier: 0,
+      };
+      roles.push(role);
+    }
+  });
 
+  const getRoleAttributes = (role: AccountRole) => {
```
[the remark this answers is not in the document]
Comment (satyajeetkolhapure) [reply src/features/account/components/AccountRoleTags.tsx:71]: I think functions are more readable, but I am fine with hashmap too

File: src/features/account/components/AccountRoleTags.tsx [hunk 2]
```diff
@@ -1,64 +1,123 @@
...
+          color: '#AE00FF',
+          label: t`Auditor`,
+        };
+    }
+  };
+
+  const getRoleIcon = (role: AccountRole) => {
+    switch (role.roleType) {
+      case AccountRoleType.Developer:
+        return '🧑‍💻';
```
[the remark this answers is not in the document]
Comment (satyajeetkolhapure) [reply src/features/account/components/AccountRoleTags.tsx:97]: Unfortunately we don't have svg for these, I got confirmation from Erik to use emoji instead. But yes svg would have been better

File: src/features/account/components/AccountInfo.tsx [hunk 3]
```diff
@@ -24,7 +25,12 @@ export const AccountInfo: FunctionComponent<AccountInfoProps> = ({
 
   return (
     <VStack spacing="8" data-testid="account-info">
-      <JazzIcon address={address} size={130} />
+      <HStack>
```
[the remark this answers is not in the document]
Comment (satyajeetkolhapure) [reply src/features/account/components/AccountInfo.tsx:28]: makes sense. I will remove it

File: src/features/account/components/AccountInfo.tsx [hunk 4]
```diff
@@ -24,7 +25,12 @@ export const AccountInfo: FunctionComponent<AccountInfoProps> = ({
...
+        <JazzIcon address={address} size={130} />
+      </HStack>
```
Comment (satyajeetkolhapure) [inline src/features/account/components/AccountInfo.tsx:30]:
```suggestion
```

File: src/features/account/components/AccountInfo.tsx [hunk 5]
```diff
@@ -24,7 +25,12 @@ export const AccountInfo: FunctionComponent<AccountInfoProps> = ({
...
+      <HStack>
```
Comment (satyajeetkolhapure) [inline src/features/account/components/AccountInfo.tsx:31]:
```suggestion
```

**Review by stanleyyconsensys — APPROVED** [review]

**Pull request merged**